### PR TITLE
Add more mvc/view features API docs

### PR DESCRIPTION
**Pull request by HaoK**
Part of https://github.com/dotnet/aspnetcore/issues/26397

### CODE REVIEW

File: src/Mvc/Mvc.ViewFeatures/src/AntiforgeryExtensions.cs [hunk 1]
```diff
@@ -10,6 +10,9 @@
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures
 {
+    /// <summary>
+    /// Static class that adds extension methods to <see cref="IAntiforgery"/>.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/AntiforgeryExtensions.cs:14]:
```suggestion
    /// Static class that adds extension methods to <see cref="IAntiforgery"/>. This class cannot be inherited.
```

Per the style recommendations [here](https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Class).

File: src/Mvc/Mvc.ViewFeatures/src/ViewEngines/ViewEngineResult.cs [hunk 2]
```diff
@@ -8,20 +8,41 @@
...
+        /// </summary>
         public IEnumerable<string> SearchedLocations { get; private set; }
 
+        /// <summary>
+        /// The <see cref="IView"/>.
+        /// </summary>
         public IView View { get; private set; }
 
+        /// <summary>
+        /// The name of the view.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/ViewEngines/ViewEngineResult.cs:31]:
```suggestion
        /// Gets or sets the name of the view.
```

Per the style recommendations [here](https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Property).

File: src/Mvc/Mvc.ViewFeatures/src/DefaultHtmlGeneratorExtensions.cs [hunk 3]
```diff
@@ -1,12 +1,27 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using Microsoft.AspNetCore.Mvc.Rendering;
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures
 {
+    /// <summary>
+    /// Static class that adds extension methods to <see cref="IHtmlGenerator"/>.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/DefaultHtmlGeneratorExtensions.cs:9]:
```suggestion
    /// Static class that adds extension methods to <see cref="IHtmlGenerator"/>. This class cannot be inherited.
```

Same as above with this and other static classes.

File: src/Mvc/Mvc.ViewFeatures/src/DefaultHtmlGeneratorExtensions.cs [hunk 4]
```diff
@@ -28,6 +43,17 @@ public static class DefaultHtmlGeneratorExtensions
             return tagBuilder;
         }
 
+        /// <summary>
+        ///
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/DefaultHtmlGeneratorExtensions.cs:47]: Looks like this needs to be filled out.

File: src/Mvc/Mvc.ViewFeatures/src/InputType.cs [hunk 5]
```diff
@@ -3,12 +3,34 @@
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures
 {
+    /// <summary>
+    /// An enum for input types.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/InputType.cs:7]:
```suggestion
    /// Specifies constants which define an input type.
```

Per the styleguide [here](https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Enum).

File: src/Mvc/Mvc.ViewFeatures/src/Rendering/FormMethod.cs [hunk 6]
```diff
@@ -3,9 +3,19 @@
 
 namespace Microsoft.AspNetCore.Mvc.Rendering
 {
+    /// <summary>
+    /// Enum representing the form method.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/Rendering/FormMethod.cs:7]:
```suggestion
    /// Specifies constants which define the form method used.
```

File: src/Mvc/Mvc.ViewFeatures/src/ViewDataDictionaryControllerPropertyActivator.cs [hunk 7]
```diff
@@ -10,6 +10,9 @@
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures
 {
+    /// <summary>
+    /// A <see cref="IControllerActivatorProvider"/> for a view data dictionary controller.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/ViewDataDictionaryControllerPropertyActivator.cs:14]:
```suggestion
    /// Represents a <see cref="IControllerActivatorProvider"/> for a view data dictionary controller.
```

**Review by captainsafia — COMMENTED** [review]
Left some comments inline per the styleguide at https://github.com/dotnet/dotnet-api-docs/wiki.

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs [hunk 8]
```diff
@@ -22,11 +22,34 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
     /// </summary>
     public class HtmlHelper : IHtmlHelper, IViewContextAware
     {
+        /// <summary>
+        /// css class name for input validation.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs:26]:
```suggestion
        /// CSS class name for input validation.
```

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs [hunk 9]
```diff
@@ -22,11 +22,34 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
...
+        /// </summary>
         public static readonly string ValidationInputCssClassName = "input-validation-error";
+
+        /// <summary>
+        /// css class name for valid input validation.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs:31]:
```suggestion
        /// CSS class name for valid input validation.
```

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs [hunk 10]
```diff
@@ -22,11 +22,34 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
...
+        /// </summary>
         public static readonly string ValidationInputValidCssClassName = "input-validation-valid";
+
+        /// <summary>
+        /// css class name for field validation error.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs:36]:
```suggestion
        /// CSS class name for field validation error.
```

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs [hunk 11]
```diff
@@ -22,11 +22,34 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
...
+        /// </summary>
         public static readonly string ValidationMessageCssClassName = "field-validation-error";
+
+        /// <summary>
+        /// css class name for valid field validation.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs:41]:
```suggestion
        /// CSS class name for valid field validation.
```

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs [hunk 12]
```diff
@@ -22,11 +22,34 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
...
+        /// </summary>
         public static readonly string ValidationMessageValidCssClassName = "field-validation-valid";
+
+        /// <summary>
+        /// css class name for validation summary errors.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs:46]:
```suggestion
        /// CSS class name for validation summary errors.
```

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs [hunk 13]
```diff
@@ -22,11 +22,34 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
...
+        /// </summary>
         public static readonly string ValidationSummaryCssClassName = "validation-summary-errors";
+
+        /// <summary>
+        /// css class name for valid validation summary.
```
Comment (captainsafia) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelper.cs:51]:
```suggestion
        /// CSS class name for valid validation summary.
```

**Review by captainsafia — APPROVED** [review]
LGTM. Few suggestions inline.

**Comment by pranavkm** [comment]
@HaoK could you retarget this to 5.0?

**Comment by HaoK** [comment]
Do you mean 6.0? master?  Isn't this already targeting release/5.0?

**Comment by pranavkm** [comment]
Sorry, yes `master` \ 6.0

**Comment by HaoK** [comment]
Rebased to master

**Comment by dougbu** [comment]
Oddly, AzDO doesn't seem to have any details information about the last aspnetcore-ci run other than it timed out after about 55 minutes ☹️

**Pull request merged**